### DISCUSSION TITLE
Referencing to other files result in `Unsolved ref(s)`

### DISCUSSION THREAD

**Discussion by Seym0n** [jy95/docusaurus-json-schema-plugin#497] [Q&A] [answered]
**Describe the bug**
When referencing to another local file, I get the error `Unsolved ref(s)`

**To Reproduce**
I created two files

docs\static\schemas\device\deviceSchema.json
```json
{
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Name of the device"
        },
        "image": {
            "type": "string",
            "description": "Path to the image asset corresponding to the device"
        },
        "position": {
            "$ref": "../array/tuples.json"
        }
    }
}
```
and

docs\static\schemas\array\tuples.json
```json
{
    "type": "array",
    "description": "Represent a street address such as ['1600','Pennsylvania','Avenue','NW']",
    "items": false,
    "prefixItems": [
      { "type": "number", "description": "The address number" },
      { "type": "string", "description": "The name of the street" },
      {
        "enum": ["Street", "Avenue", "Boulevard"],
        "description": "The type of street"
      },
      {
        "enum": ["NW", "NE", "SW", "SE"],
        "description": "The city quadrant of the address"
      }
    ]
  }
```

and other tests where the referenced JSON file is in the same directory using path `tuples.json`

**Expected behavior**
Normally, it should reference to the tuples and display its scheme when toggling `position`.

**Screenshots:**
![image](https://github.com/user-attachments/assets/490e3fe0-9467-4722-9987-0bde0e141a9e)

**Desktop (please complete the following information):**
 - Chrome 131.0.6778.86

**Additional context**
The version of docusaurus json plugin I use is ^1.12.2.
I also use docusaurus version 3.6.3

**Comment by jy95** [comment]
Hello @Seym0n , check the docs - I wrote an example long time ago which works and handle such case  : https://jy95.github.io/docusaurus-json-schema-plugin/docs/demo-viewer/refs/localFile ;)

**Reply by Seym0n** [reply, marked as answer]
I checked this out before, but now I noticed I missed that resolverOptions was relevant to set up. I thought it was supported out-of-the box. Ended up copying the files from here https://github.com/jy95/docusaurus-json-schema-plugin/tree/main/testsite/src/components/shared-libs

Thank you!

**Reply by jy95** [reply]
You are welcome. The thing with `resolverOptions` is that you can customize to your likings. The dynamic import with some path rewritting way did the trick for me, likely someone else has another solution
